bin/p2-stop/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/square/p2/pkg/auth"
+	"github.com/square/p2/pkg/constants"
+	"github.com/square/p2/pkg/p2stop"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/version"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const (
+	fullDescription = `Safely disable and stop an existing pod.
+
+EXAMPLES
+
+$ p2-stop mypod
+
+$ p2-stop --pod-dir /custom/pod/home
+
+`
+)
+
+var (
+	app = kingpin.New("p2-stop", fullDescription)
+
+	nodeName = app.Flag("node-name", "The name of this node (default: hostname)").String()
+	podDir   = app.Flag("pod-dir", "The directory where the pod to be halted is located. ").String()
+	podName  = app.Arg("pod-name", fmt.Sprintf("The name of the pod to be halted. Looks in the default pod home '%s' for the pod", pods.DefaultPath)).String()
+)
+
+func main() {
+	app.Version(version.VERSION)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+
+	pods.Log.Logger.Formatter = &logrus.TextFormatter{
+		DisableTimestamp: false,
+		FullTimestamp:    true,
+		TimestampFormat:  "15:04:05.000",
+	}
+	logger := pods.Log
+
+	if p2stop.PolicyPath == "" || p2stop.Keyring == "" || p2stop.DefaultDomain == "" {
+		logger.Fatalln("PolicyPath or Keyring or DefaultDomain need to be set at build time")
+	}
+
+	if *nodeName == "" {
+		hostname, err := os.Hostname()
+		if err != nil {
+			logger.WithError(err).Fatalln("Error getting node name")
+		}
+		*nodeName = hostname
+	}
+
+	if *podName == "" && *podDir == "" {
+		logger.NoFields().Fatalln("Must pass a pod name or pod home directory")
+	}
+
+	var path string
+	if *podName != "" {
+		path = filepath.Join(pods.DefaultPath, *podName)
+	} else {
+		path = *podDir
+	}
+
+	// TODO: pass requirefile from old pod to new pod
+	pod, err := pods.PodFromPodHome(types.NodeName(*nodeName), path)
+	if err != nil {
+		logger.NoFields().Fatalln(err)
+	}
+
+	manifest, err := pod.CurrentManifest()
+	if err != nil {
+		logger.NoFields().Fatalln(err)
+	}
+
+	authPolicy, err := auth.NewUserPolicy(
+		p2stop.Keyring,
+		p2stop.PolicyPath,
+		constants.PreparerPodID,
+		constants.PreparerPodID.String(),
+	)
+	if err != nil {
+		logger.WithError(err).Fatalln("error configuring user auth")
+	}
+
+	user := os.Getenv("SUDO_USER")
+	if user == "" {
+		logger.Fatalln("Could not get SUDO_USER")
+	}
+	email := user + p2stop.DefaultDomain
```

Comment:
golang prefers fmt.Sprintf() for building strings. I also suspect you need an "@" in here somewhere

Reply:
The default domain in the build script was set to "@squareup.com".